### PR TITLE
fix(crons): Ignore muted status when running `mark_ok`

### PR DESCRIPTION
**Pull request by wedamija**
If a monitor is in a bad state, we don't want to leave it stuck there just because it's muted - it should still be able to get resolved and close out the related issue, if one exists.

### CODE REVIEW

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 1]
```diff
@@ -722,7 +722,7 @@ def test_mark_failed_issue_threshold(self, mock_produce_occurrence_to_kafka):
 
         # check that incident has not changed
         monitor_incident = MonitorIncident.objects.get(id=monitor_incident.id)
-        assert monitor_incident.grouphash == monitor_environment.incident_grouphash
+        assert monitor_incident.grouphash == monitor_environment.active_incident
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:725]:
```suggestion
        assert monitor_incident.grouphash == monitor_environment.active_incident.grouphash
```

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 2]
```diff
@@ -830,7 +830,7 @@ def test_mark_failed_issue_threshold_timeout(self, mock_produce_occurrence_to_ka
         monitor_incident = monitor_incidents.first()
         assert monitor_incident.starting_checkin == first_checkin
         assert monitor_incident.starting_timestamp == first_checkin.date_added
-        assert monitor_incident.grouphash == monitor_environment.incident_grouphash
+        assert monitor_incident.grouphash == monitor_environment.active_incident
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:833]:
```suggestion
        assert monitor_incident.grouphash == monitor_environment.active_incident.grouphash
```

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 3]
```diff
@@ -698,7 +698,7 @@ def test_mark_failed_issue_threshold(self, mock_produce_occurrence_to_kafka):
...
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:701]:
```suggestion
        assert monitor_incident.grouphash == monitor_environment.active_incident.grouphash
```

**Review by evanpurkhiser — APPROVED** [review]
Couple test issues but looks good to go

File: src/sentry/monitors/logic/mark_ok.py [hunk 4]
```diff
@@ -52,24 +40,14 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
...
-                resolving_checkin=checkin,
-                resolving_timestamp=checkin.date_added,
-            )
-        else:
-            # Don't update status if incident isn't recovered
-            params.pop("status", None)
+            params["status"] = MonitorStatus.OK
+            incident = monitor_env.active_incident
+            if incident:
+                resolve_incident_group(incident.grouphash, checkin.monitor.project_id)
```
Comment (davidenwang) [inline src/sentry/monitors/logic/mark_ok.py:46]: niceee glad we have the guarantee of only one open incident at a time now

**Review by davidenwang — APPROVED** [review]

**Pull request merged**